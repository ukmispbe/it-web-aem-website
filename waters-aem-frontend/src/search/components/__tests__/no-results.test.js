import React from 'react';
import renderer from 'react-test-renderer';
import NoResults from '../no-results';

describe('Feature: NoResults React Component', () => {
    describe('Scenario: Rendering', () => {
        describe('When the component renders with props', () => {
            it('Then the snapshot should match', () => {
                const searchText = {
                    noResultsTitle: 'TITLE',
                    noResultsDescription: 'DESCRIPTION',
                    noResultsSearchLinkText: 'LINK_TEXT'
                };
                const query = "QUERY";
                const json = renderer.create(<NoResults searchText={searchText} query={query} />);

                expect(json).toMatchSnapshot();
            });
        });
    });
});