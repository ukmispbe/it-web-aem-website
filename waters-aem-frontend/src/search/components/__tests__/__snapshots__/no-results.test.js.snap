// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: NoResults React Component Scenario: Rendering When the component renders with props Then the snapshot should match 1`] = `
Array [
  <h2
    className="cmp-search__resultsCount noresults"
  >
    Showing 0 results
     
     
    <span>
      for "
      <strong>
        QUERY
      </strong>
      "
    </span>
  </h2>,
  <div
    className="cmp-search__no-results"
  >
    <div
      className="icon"
    />
    <h2>
      TITLE
    </h2>
    <p>
      DESCRIPTION
      <a
        href="http://localhost/"
      >
        LINK_TEXT
      </a>
    </p>
  </div>,
]
`;
